castra/core.py

```diff
@@ -65,23 +74,37 @@ def __init__(self, path=None, template=None):
     def load_meta(self, loads=pickle.loads):
         meta = []
         for name in ['columns', 'dtypes', 'index_dtype']:
-            with open(join(self.meta_path, name), 'r') as f:
+            with open(self.dirname('meta', name), 'r') as f:
                 meta.append(loads(f.read()))
         self.columns, self.dtype, self.index_dtype = meta
 
     def flush_meta(self, dumps=pickle.dumps):
         for name in ['columns', 'dtypes', 'index_dtype']:
-            with open(join(self.meta_path, name), 'w') as f:
+            with open(self.dirname('meta', name), 'w') as f:
                 f.write(dumps(getattr(self, name)))
 
     def load_partitions(self, loads=pickle.loads):
-        with open(join(self.meta_path, 'plist'), 'r') as f:
+        with open(self.dirname('meta', 'plist'), 'r') as f:
             self.partitions = pickle.loads(f.read())
 
     def save_partitions(self, dumps=pickle.dumps):
-        with open(join(self.meta_path, 'plist'), 'w') as f:
+        with open(self.dirname('meta', 'plist'), 'w') as f:
             f.write(dumps(self.partitions))
 
+    def append_categories(self, new):
+        for col, cat in new.items():
+            if cat:
+                with open(self.dirname('meta', 'categories', col), 'a') as f:
```

Comment:
Could this present problems if there's a write failure? It will write some of the new categories but not all. I think we want this operation to be all or none.